cpp/src/gandiva/cache.h

```diff
@@ -25,6 +25,13 @@
 
 namespace gandiva {
 
+namespace internal {
+// Only called once by GetCapacity().
+// Do the actual work of getting the capacity from env var.
+// Also makes the testing easier.
```

Comment:
It is not possible in google test to re-initialize a static variable. So have this dedicated function to do the actual work eagerly, then we can test it instead of `GetCapacity` (which contains the static variable).